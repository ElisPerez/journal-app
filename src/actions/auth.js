// AquÃ­ van todas las actions
// Nota: Snippet para una export const: enf

import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types';

// action login with Google
export const startGoogleLogin = () => {
	return (dispatch) => {
		firebase
			.auth()
			.signInWithPopup(googleAuthProvider)
			// .then( userCredential => {
				// console.log(userCredential);
			.then(({ user }) => {
				dispatch(
						login(user.uid, user.displayName)
					)
				console.log(user.uid);
				console.log(user.displayName);
				console.log(user.email);
				console.log(user.photoURL);
				// user.uid
				// user.displayName
				// user.email
				// user.photoURL
			});
	};
};

// action login:
export const login = (uid, displayName) => ({
	type: types.login,
	payload: {
		uid,
		displayName,
	},
});

// action login with email and password type async example:
export const startLoginEmailPassword = (email, password) => {
	return (dispatch) => {
		setTimeout(() => {
			dispatch(login(123, 'Pedrito'));
		}, 2000);
	};
};
