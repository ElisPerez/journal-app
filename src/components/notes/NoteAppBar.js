import React from 'react';

export const NoteAppBar = () => {
	return (
		<div className='notes__appbar'>
   <span>23 de agosto 2983</span>
   <div>
    <button className='btn'>
     Picture
    </button>
    <button className='btn'>
     Save
    </button>
   </div>
		</div>
	);
};
