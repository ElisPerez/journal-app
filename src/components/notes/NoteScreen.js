import React, { useEffect, useRef } from 'react';
import { useSelector } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { NoteAppBar } from './NoteAppBar';

export const NoteScreen = () => {
	const { active: note } = useSelector((state) => state.notes);
	const [formValues, handleInputChange, reset] = useForm(note);
	const { body, title } = formValues;
	const activeId = useRef(note.id);

	useEffect(() => {
		if (note.id !== activeId.current) {
			reset(note);
			activeId.current = note.id;
		}
	}, [note, reset]);

	return (
		<div className='notes__main-content'>
			<NoteAppBar />
			<div className='notes__content'>
				<input
					autoComplete='off'
					className='notes__title-input'
					name='title'
					onChange={handleInputChange}
					placeholder='Some awesome title'
					type='text'
					value={title}
				/>
				<textarea
					className='notes__textarea'
					name='body'
					onChange={handleInputChange}
					placeholder='What happened today'
					value={body}
					cols='30'
					rows='10'
				></textarea>
				{/* <div className='notes__image'> NO FUNCIONA ASI. TUVE QUE COLOCAR LA CLASE EN LA TAG IMG PERO DENTRO DE UN DIV SIN CLASES */}
				{note.url && (
					<div>
						<img
							className='notes__image'
							src='https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg'
							alt='landscape'
						/>
					</div>
				)}
			</div>
		</div>
	);
};
