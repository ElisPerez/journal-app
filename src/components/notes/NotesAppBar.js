import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
dayjs.extend(advancedFormat);

export const NotesAppBar = () => {
	const { active } = useSelector((state) => state.notes);
	const dateActiveNote = active.date;

	const dispatch = useDispatch();
	const handleSave = () => {
		dispatch(startSaveNote(active));
	};
	const handlePictureClick = () => {
		document.querySelector('#fileSelector').click();
	};
	const handleFileChange = (e) => {
		const file = e.target.files[0];
		if (file) {
			dispatch(startUploading(file));
		}
	};
	return (
		<div className='notes__appbar'>
			<span>
				Note made on {dayjs(dateActiveNote).format('MMMM Do YYYY')} at{' '}
				{dayjs(dateActiveNote).format('hh:mm:ssa')}
			</span>
			<input
				id='fileSelector'
				name='file'
				onChange={handleFileChange}
				style={{ display: 'none' }}
				type='file'
			/>
			<div>
				<button className='btn' onClick={handlePictureClick}>
					Picture
				</button>
				<button className='btn' onClick={handleSave}>
					Save
				</button>
			</div>
		</div>
	);
};
