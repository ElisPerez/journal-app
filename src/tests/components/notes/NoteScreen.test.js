import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {
    uid: '1',
    name: 'Elis Antonio',
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: 1234,
      title: 'Hello',
      body: 'word',
      date: 0,
    },
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);

describe('Testing in the <NoteScreen />', () => {
  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  }); // test 1

  test('should dispatch the active note', () => {
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'Hello again',
      },
    });

    expect(activeNote).toHaveBeenLastCalledWith(
        1234,
        {
            body: 'word',
            title: 'Hello again',
            id: 1234,
            date: 0,
        }
    )  
  }); // test 2
}); // describe
