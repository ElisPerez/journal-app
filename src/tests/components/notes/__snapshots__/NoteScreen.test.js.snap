// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in the <NoteScreen /> should show correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      Note made on 
      December 31st 1969
       at
       
      07:00:00pm
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hello"
    />
    <textarea
      className="notes__textarea"
      cols="12"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      rows="10"
      value="word"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
